wallet/wallet.go

```diff
@@ -493,6 +506,85 @@ func (w *Wallet) waitUntilBackendSynced(chainClient chain.Interface) error {
 	}
 }
 
+// locateBirthdayBlock returns a block that meets the given birthday timestamp
+// by a margin of +/-2 hours. This is safe to do as the timestamp is already 2
+// days in the past of the actual timestamp.
+func locateBirthdayBlock(chainClient chain.Interface,
+	birthday time.Time) (*waddrmgr.BlockStamp, error) {
+
+	// Retrieve the lookup range for our block.
+	startHeight := int32(0)
+	_, bestHeight, err := chainClient.GetBestBlock()
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugf("Locating suitable block for birthday %v between blocks "+
+		"%v-%v", birthday, startHeight, bestHeight)
+
+	var (
+		birthdayBlock *waddrmgr.BlockStamp
+		left, right   = startHeight, bestHeight
+	)
+
+	// Binary search for a block that meets the birthday timestamp by a
```

Comment:
I think we need to be careful here, since header timestamps are not always increasing. Could do something like always jump 11 blocks back to be sure.

Reply:
The wallet already sets its birthday to two days before the birthday in the seed, so we already have enough breathing room I think?

Reply:
> The wallet already sets its birthday to two days before the birthday in the seed, so we already have enough breathing room I think?

Yes I think this is fine as well.

Reply:
Yeah, then we should be fine.